From 334fec9d3074e82b7bf343871e066c18cc66b4a1 Mon Sep 17 00:00:00 2001
From: Benedikt Niedermayr <Benedikt.Niedermayr@mixed-mode.de>
Date: Wed, 14 Jun 2017 16:42:45 +0200
Subject: [PATCH 4/5] spi-sun6i: Fixed maximum transfer size of 64bit.

The standart spi-sun6.c is only capable of transmitting 64 bits which is the SPI FIFO length.
But the SPI interface can also refill the FIFO, when transmitting larger messages as 64 bits.

Signed-off-by: Benedikt Niedermayr <Benedikt.Niedermayr@mixed-mode.de>
---
 drivers/spi/spi-sun6i.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-sun6i.c b/drivers/spi/spi-sun6i.c
index 88663ff..7c80b37 100644
--- a/drivers/spi/spi-sun6i.c
+++ b/drivers/spi/spi-sun6i.c
@@ -217,6 +217,7 @@ static int sun6i_spi_transfer_one(struct spi_master *master,
 	int ret = 0;
 	u32 reg;
 
+	/* We don't support transfer larger than the FIFO */
 	if (tfr->len > SUN6I_MAX_XFER_SIZE)
 		return -EINVAL;
 
-- 
2.7.4

